### PR TITLE
add spread() to kew

### PR DESCRIPTION
**Pull request by nicks**
Hello @kylehg, 

Please review the following commits I made in branch 'nick-spread'.

9ad3424f51cc6dafa29d9f26a10f9b40984dc0e8 (2015-05-13 18:55:25 -0400)
add spread() to kew

R=@kylehg

### CODE REVIEW

File: test/spread.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
+var Q = require('../kew')
+var originalQ = require('q')
```
Comment (kylehg) [inline test/spread.js]: :wave:

File: kew.js [hunk 2]
```diff
@@ -299,6 +299,38 @@ Promise.prototype.failBound = function (onFail, scope, var_args) {
...
 /**
+ * Spread a promises outputs to the functions arguments.
+ * @param {?function(this:void, ...): RESULT|undefined} onSuccess
+ * @return {!Promise.<RESULT>} returns a new promise with the output of the onSuccess or
+ *     onFail handler
+ * @template RESULT
+ */
+Promise.prototype.spread = function (onSuccess) {
+  return this
+  .then(allInternal)
```
Comment (kylehg) [inline kew.js]: Can you put this on the same line as `this`? I find this construction very hard to parse. Ditto below

**Comment by kylehg** [comment]
This is awesome, @nicks! I was just thinking how we should have the equivalent of [RSVP.hash()](https://github.com/tildeio/rsvp.js/#hash-of-promises), but this is even better

**Comment by kylehg** [comment]
I don't understand the Travis failures, but lgtm

File: test/spread.js (hunk 1 above)
Comment (nicks) [inline test/spread.js]: removed

File: kew.js (hunk 2 above)
Comment (nicks) [inline kew.js]: done

Comment (chaosgame) [inline kew.js]: Internally, we're using `setImmediate` for our nextTick function.  Internally, when we call `setImmediate`, Node puts the function on a queue.  Each time we cycle through the event loop, Node pops a single function off the queue.

If we front-load each item in the array onto the queue, we're going to execute them all on sequential executions of the event loop.

A more fair way of implementing this would be to have `allInternal` run in an asynchronous loop, where we execute `onSuccess` for the first element, and then call `setImmediate` to enqueue our next iteration through the array.

This isn't an issue for small arrays, but it could lead to more difficult to predict behavior for larger arrays.

**Comment by nicks** [comment]
we're going to have a separate pr to discuss nathan's concerns about Q.all
